src/dmd/cpreprocess.d

```diff
@@ -50,42 +50,39 @@ FileName preprocess(FileName csrcfile, out bool ifile)
         const ext = FileName.ext(name);
         assert(ext);
         const ifilename = FileName.addExt(name[0 .. name.length - (ext.length + 1)], i_ext);
-        auto status = runPreprocessor(cppCommand(), csrcfile.toString(), ifilename);
+        const command = cppCommand();
+        auto status = runPreprocessor(command, csrcfile.toString(), ifilename);
         if (status)
         {
-            error(Loc.initial, "C preprocess failed for file %s, exit status %d\n", csrcfile.toChars(), status);
+            error(Loc.initial, "C preprocess command %.*s failed for file %s, exit status %d\n",
+                cast(int)command.length, command.ptr, csrcfile.toChars(), status);
             fatal();
         }
         ifile = true;
         return FileName(ifilename);
     }
     else version (Windows)
     {
-        /* This actually works, but fails the test suite because:
-           1. For Microsoft's preprocessor, it insists on printing the filename it's preprocessing,
-              which causes all the fail_compilation tests to fail. Yes, even with /nologo.
-              cl.exe does not appear to have a "quiet" mode.
-           2. For Win32 with Digital Mars, the tests all fail because sppn.exe, the DMC preprocessor,
-              is not on the path. Even if it was, it would still fail because the DMC headers are not
-              there. To get the headers and sppn.exe, dmc will need to be installed on the test machines:
-              http://ftp.digitalmars.com/Digital_Mars_C++/Patch/dm857c.zip
+        /*
+           To get sppn.exe: http://ftp.digitalmars.com/sppn.zip
+           To get the dmc C headers, dmc will need to be installed:
+           http://ftp.digitalmars.com/Digital_Mars_C++/Patch/dm857c.zip
          */
-        if (target.objectFormat() == Target.ObjectFormat.coff)
+        const name = FileName.name(csrcfile.toString());
+        const ext = FileName.ext(name);
+        assert(ext);
+        const ifilename = FileName.addExt(name[0 .. name.length - (ext.length + 1)], i_ext);
+        const command = cppCommand();
+        auto status = runPreprocessor(command, csrcfile.toString(), ifilename);
```

Comment:
`preprocess` now has the same code for Posix and Windows. Folding that into unconditional code and keeping OS differences to `runPreprocessor` seems natural.

Reply:
It's a good idea, but I'd prefer to do that as a refactoring.